functions/firebase/firestore-reactive/src/main/java/functions/FirebaseFirestoreReactive.java

```diff
@@ -79,11 +79,17 @@ public void accept(String json, Context context) {
 
     String affectedDoc = context.resource().split("/documents/")[1].replace("\"", "");
 
-    logger.info(String.format("Replacing value: %s --> %s", currentValue, newValue));
-    try {
-      FIRESTORE.document(affectedDoc).set(newFields, SetOptions.merge()).get();
-    } catch (ExecutionException | InterruptedException e) {
-      logger.log(Level.SEVERE, "Error updating Firestore document: " + e.getMessage(), e);
+    if (!currentValue.equals(newValue)) {
```

Comment:
Consider a comment about why you are doing this.  Something as simple as "performance optimization" will do.

Reply:
Is the comment in the `else` clause enough?

Reply:
Since it's kind of important, it makes sense to say it early so they understand why the if is there.

Reply:
Comments reworded, @lesv PTAL?